File: Moon.Script/MarkerFlipSync.fs
namespace Moon

open Godot

[<FScript(typeof<MarkerFlipSync>)>]
type private MarkerFlipSyncScript (marker : MarkerFlipSync) =

    let physics = marker.ProcessCallback = MarkerFlipSync.MarkerFlipSyncProcessCallback.Physics
    let anim = marker |> Node.tryGetNode marker.Target

    let update () =
        anim
        |> Option.filter GodotObject.IsInstanceValid
        |> Option.iter (fun a ->
            marker.FlipH <- a |> Flip.getH
            marker.FlipV <- a |> Flip.getV
            marker.Position <- marker.Origin.Flip(marker.FlipH, marker.FlipV)
        )

    do

        marker |> Node.whenReady (fun () ->
            marker.Origin <- marker.Position
            update ()
        )

        marker |> Engine.addProcess physics update |> ignore

    [<GDMember>]
    member this.Update() = update()

File: Moon.Class/Component/MarkerFlipSync.cs
using System;
using Godot;

namespace Moon.Component;

[GlobalClass]
public partial class MarkerFlipSync : Marker2D
{
    public enum MarkerFlipSyncProcessCallback { Idle, Physics }
    
    [Export(PropertyHint.NodePathValidTypes, "Sprite2D,AnimatedSprite2D")]
    public NodePath Target { get ;set; } = "..";
    
    [Export]
    public MarkerFlipSyncProcessCallback ProcessCallback { get; set; } 
        = MarkerFlipSyncProcessCallback.Physics;
    
    public bool FlipH { get; set; } = false;
    public bool FlipV { get; set; } = false;
    public Vector2 Origin { get; set; }
    
    public event Action Updated;
    public void Update() => Updated?.Invoke();
}
